src/flatten_dict/tests/flatten_dict_test.py

```diff
@@ -60,6 +72,21 @@ def test_flatten_dict(normal_dict, flat_tuple_dict):
     assert flatten(normal_dict) == flat_tuple_dict
 
 
+def test_flatten_dict_depth_limit(normal_dict, flat_tuple_dict_depth2):
+    assert flatten(normal_dict, max_depth=2) == flat_tuple_dict_depth2
+
+
+def test_flatten_dict_irrelevant_depth_limit(normal_dict, flat_tuple_dict):
+    assert flatten(normal_dict, max_depth=3) == flat_tuple_dict
+
+
+def test_flatten_dict_zero_depth_limit(normal_dict):
+    flattened = flatten(normal_dict, max_depth=0)
+    values_before = sorted(flattened.values(), key=lambda x: json.dumps(x, sort_keys=True))
+    values_after = sorted(normal_dict.values(), key=lambda x: json.dumps(x, sort_keys=True))
+    assert values_before == values_after
```

Comment:
why not simply `flattened == normal_dict`?

Reply:
See my comment for an explanation